server/controllers/temporary-accommodation/manage/premisesController.ts

```diff
@@ -119,9 +119,16 @@ export default class PremisesController {
       const { premisesId } = req.params
       const callConfig = extractCallConfig(req)
 
+      const { name } = req.body
+
+      const premises = await this.premisesService.getPremises(callConfig, premisesId)
+
+      const newPremisesName = name === premises.name ? null : name
+
       const updatePremises: UpdatePremises = {
         characteristicIds: [],
         ...req.body,
+        name: newPremisesName,
```

Comment:
A shame we have to go and query it first isn't it.
Presumably we can't send whatever `name` is and let the API return 422 (In our case I think it 400s for some reason)?

Reply:
Yeah it's not ideal. I think this logic should live in the [API](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PremisesService.kt#L446).

I can make an API card for it?